pkg/registry/core/pod/storage/storage.go

```diff
@@ -180,20 +181,24 @@ func (r *BindingREST) Create(ctx context.Context, name string, obj runtime.Objec
 		}
 	}
 
-	err = r.assignPod(ctx, binding.Name, binding.Target.Name, binding.Annotations, dryrun.IsDryRun(options.DryRun))
+	err = r.assignPod(ctx, binding.UID, binding.Name, binding.Target.Name, binding.Annotations, dryrun.IsDryRun(options.DryRun))
 	out = &metav1.Status{Status: metav1.StatusSuccess}
 	return
 }
 
 // setPodHostAndAnnotations sets the given pod's host to 'machine' if and only if it was
 // previously 'oldMachine' and merges the provided annotations with those of the pod.
 // Returns the current state of the pod, or an error.
-func (r *BindingREST) setPodHostAndAnnotations(ctx context.Context, podID, oldMachine, machine string, annotations map[string]string, dryRun bool) (finalPod *api.Pod, err error) {
+func (r *BindingREST) setPodHostAndAnnotations(ctx context.Context, podUID types.UID, podID, oldMachine, machine string, annotations map[string]string, dryRun bool) (finalPod *api.Pod, err error) {
 	podKey, err := r.store.KeyFunc(ctx, podID)
 	if err != nil {
 		return nil, err
 	}
-	err = r.store.Storage.GuaranteedUpdate(ctx, podKey, &api.Pod{}, false, nil, storage.SimpleUpdate(func(obj runtime.Object) (runtime.Object, error) {
+	preCondition := &storage.Preconditions{}
+	if podUID != "" {
+		preCondition.UID = &podUID
```

Comment:
if we're adding support for UID precondition, it would make sense to also support resourceVersion precondition if set

Reply:
got it. resourceVersion precondition added in new commit.

Reply:
> if we're adding support for UID precondition, it would make sense to also support resourceVersion precondition if set

got it. precondition for resourceVersion added in new commit.

Reply:
or keep the UID precondition and resourceVersion precondition, scheduler should divide schedule process into two step: 1. run algorithm to get node; 2. call binding api; if the step 2 check the UID not change, retry.
the purpose is: don't call schedule algorithm twice, because it's expensive

Reply:
this code honors the uid and resourceVersion preconditions *if set*

for the scheduler's purposes, I would only expect it to make use of the uid precondition